ConfigSpace/hyperparameters.pyx

```diff
@@ -358,6 +368,21 @@ cdef class IntegerHyperparameter(NumericalHyperparameter):
                              " %s." % (name, type(parameter), str(parameter)))
         return int(parameter)
 
+    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[np.ndarray, float, int]]:
+        if isinstance(vector, np.ndarray):
```

Comment:
Is there a reason to not use `_check_vector` here?

Reply:
Same as above.